dev/com.ibm.ws.security.authorization.jacc.ejb/src/com/ibm/ws/security/authorization/jacc/ejb/impl/EJBSecurityValidatorImpl.java

```diff
@@ -84,67 +92,81 @@ private boolean checkMethodConstraints(final String contextId,
-                                } catch (NamingException e) {
-                                    if (tc.isDebugEnabled())
-                                        Tr.debug(tc, "NamingException is caught. Ignoring..", e);
-                                }
-                                if (mc != null) {
-                                    if (tc.isDebugEnabled())
-                                        Tr.debug(tc, "javax.xml.soap.SOAPMessage is set: ", mc);
-                                    handlerObjects.put(jaccHandlerKeyArray[3], mc);
-                                }
-                                ProtectionDomain pd = null;
-
-                                if (subject != null && subject.getPrincipals().size() > 0) {
-                                    Principal[] principalArray = subject.getPrincipals().toArray(new Principal[subject.getPrincipals().size()]);
-                                    pd = new ProtectionDomain(nullCs, null, null, principalArray);
-                                } else {
-                                    pd = nullPd;
-                                }
-
-                                if (tc.isDebugEnabled())
-                                    Tr.debug(tc, "Setting JACC handler data");
-                                PolicyContext.setHandlerData(handlerObjects);
-                                if (tc.isDebugEnabled())
-                                    Tr.debug(tc, "Calling JACC implies. PD : " + pd);
-                                return Policy.getPolicy().implies(pd, permission);
-                            }
-                        });
+                                               new PrivilegedExceptionAction<Boolean>() {
+                                                   @Override
+                                                   public Boolean run() throws javax.security.jacc.PolicyContextException {
+                                                       PolicyContext.setContextID(contextId);
+
+                                                       if (tc.isDebugEnabled())
+                                                           Tr.debug(tc, "Registering JACC context handlers");
+                                                       for (String key : jaccHandlerKeyArray) {
+                                                           PolicyContext.registerHandler(key, pch, true);
+                                                       }
+                                                       handlerObjects.put(jaccHandlerKeyArray[0], subject);
+                                                       handlerObjects.put(jaccHandlerKeyArray[1], bean);
+                                                       handlerObjects.put(jaccHandlerKeyArray[2], methodParameters);
+
+                                                       /*
+                                                        * JavaEE supports JAX-RPC MessageContext. JakartaEE removed this support.
```

Comment:
To be clear JakartaEE 9 removed this support.  It is still there in JakartaEE 8

Reply:
I had hoped that the version check would have been here before newing up the InitialContext.

Reply:
I moved the version check so we don't create the context and clarified the comment.